fg/apps/factory/urls.py

```diff
@@ -14,11 +14,18 @@
 
 urlpatterns = [
     url(r'^$', views.factory_view, name='factory'),
-    url(r'^twist/plates/import$', views.twist_import_plates, name='twist_import_plates'),
-    url(r'^twist/parts/import$', views.twist_import_parts, name='twist_import_parts'),
-    url(r'^factoryorder/parts/completed/(?P<uuid>.+)$', views.view_factoryorder_parts_completed, name='view_factoryorder_parts_completed'),
-    url(r'^factoryorder/parts/failed/(?P<uuid>.+)$', views.view_factoryorder_parts_failed, name='view_factoryorder_parts_failed'),
-    url(r'^factoryorder/parts/(?P<uuid>.+)$', views.view_factoryorder_parts, name='view_factoryorder_parts'),
+
+    # Twist
+    url(r'^twist/plates/import/?$', views.twist_import_plates, name='twist_import_plates'),
+    url(r'^twist/parts/import/?$', views.twist_import_parts, name='twist_import_parts'),
+
+    # Bionet Server Import
+    url(r'^factory/plate/import/?$', views.import_factory_plate, name='import_factory_plate'),
```

Comment:
So just to be clear - if someone exports a distribution or a plateset, you want absolutely no record of the distribution or plateset, just the isolated plates?

Inventory --> user facing views 
Factory --> lab facing views

If I'm a lab admin and I would want to import something, my expectation is that it would be alongside all the other buttons that allow me to import (from Shippo or Twist). If we put these in very different places, the interface becomes non-intuitive.

Reply:
> So just to be clear - if someone exports a distribution or a plateset, you want absolutely no record of the distribution or plateset, just the isolated plates?

No, I'd like there to be a record of the distribution or plate set. However, I don't want to export every single plate from each one of those per exportation because that physically wouldn't make sense (we're not shipping all of our plates to people, we're shipping a subset of the plates to them, so their system should import a subset of the plates)

> If we put these in very different places, the interface becomes non-intuitive.

Ok, I agree with that stance. We can keep it in Factory.

Reply:
okay I misunderstood - so you do want to export a record of distribution or plateset, but you want another page entirely to select a subset of distribution plates? E.g., this part:

> Distributions / Platesets should only export the plates that were physically sent (ie, accepts a list of plate UUIDS)

That makes the view much more complicated, because we can't just add a button, we need to add a new form, redirect, etc.

Reply:
> so you do want to export a record of distribution or plateset, but you want another page entirely to select a subset of distribution plates?

I think I only really want the second (distributions / platesets are purposeless without their plates). 

> That makes the view much more complicated, because we can't just add a button, we need to add a new form, redirect, etc.

What would be the simplest way to implement this? And a second question that may be out of scope: in the older API, Plates are tracked by shipments. However, in the newer system, I couldn't find a shipment model - are Plates that are shipped not tracked?

Reply:
So - you _don't_ want to export a record of a distribution or plateset, but you _do_ want a page to select plates from a distribution or plateset (and export them as just plates?). I'm really having trouble understanding what you are asking for, please describe it clearly.

Reply:
There is no shipment model, a shipment is an Order object.

Reply:
We want to be able to export the records of a Distribution / associated plateset, but with the caveat that we *only* want that IF we can export a subset of their associated plates, and not every plate associated with each one.

Reply:
okay that's what I thought you originally said, so why did you say

> I think I only really want the second (distributions / platesets are purposeless without their plates).

to imply that you don't want what I said first, which is to export a record of distribution/plateset?

Reply:
> to imply that you don't want what I said first, which is to export a record of distribution/plateset?

Because I assumed "page entirely to select a subset of distribution plates?" meant the distribution as well as the plate, because otherwise, it's not a subset of distribution plates, it's just a subset of plates.

Reply:
From a modeling perspective, this isn't logical to me. Grouped sets of known plates would coincide with platesets - and then the only reason for a Distribution object to exist is to package some number of platesets. To "export a Distribution" but only include a subset of plates isn't really representing the export of the distribution, it's a subset of plates for it, which would be a distribution. If you are exporting plates from a distribution and want to retain some record of where the plate came from, that shouldn't mean trying to reproduce the model (but differently) in a different node, it should be the case that the unique id for the plate links back to the distribution in the previous node. What you're creating here are duplicated distributions that have different sets of plates. It's just not logical, it feels like it's going to become very difficult to manage down the line.

My 0.02.

Reply:
So here is me from a user perspective:

I've just got a set of plates from EndyLab. They sent me 3 copies of the OpenEnzyme set. I'd like to redistribute some of these plates, so I download the OpenEnzyme distribution with this subset of plates for my node, and boom, there are distributions someone can request on the site. 

From a modeling perspective of what is going on in real world, I can't see how exporting all Plates associated with a distribution makes sense - those physicals ain't in your node.

It is sort of similar to virtuals and physicals. You could have all the FreeGenes Parts in your Bionet Node and zero samples. In the case of physicals, we only want things in the physical world to be represented - while with virtuals, we're happy replicating them across however many nodes are needed.

It is ok for a Part to have a confirmed Sample on one server, but a different confirmed Sample on a different server. Similarly, it is ok for a Distribution / Plateset set to have different plates on each server, so long as each one of those plates represent the same Parts.

Reply:
okay, so now let's say I'm a third user, and I'm looking at FreeGenes and Bionet Node Number 2. I find two distributions with the same identifier and name, but the plates are different. Which one do I request? How do I know which one is the "correct" distribution? If a distribution does not need to represent a hardened set, and it's a constantly moving target that has different plates across nodes (and I don't care which one I get) then this makes sense. Otherwise, :/

Reply:
> Which one do I request? 

I find the same headphones on eBay and Amazon. Which do I order?

> How do I know which one is the "correct" distribution?

Theoretically, all should be the correct distribution. In practice, it'll probably come down to the community to not use the same name for multiple distributions. If there a malicious actor (or ignorant actor), they can likely just be flagged by the community.

> If a distribution does not need to represent a hardened set

A distribution already isn't really a hardened set, which is the reason why platesets have a many-to-many relationship with distributions instead of many-to-one. Plates can be freely swapped between many different distributions (in theory, in practice this probably won't happen too often).

Reply:
For the headphones example, you are getting the same product just from two different distributions. That is different from here, where you have things labeled as equal that have different contents.

> Theoretically, all should be the correct distribution. In practice, it'll probably come down to the community to not use the same name for multiple distributions. If there a malicious actor (or ignorant actor), they can likely just be flagged by the community.

okay, so should we not be importing distribution ids or names then? How does one import a distribution without keeping record of the name and uuid?

Reply:
> For the headphones example, you are getting the same product just from two different distributions.

In that example, you're getting the same virtual (product) from two different distributions (each sending you different atoms).

> That is different from here, where you have things labeled as equal that have different contents.

The headphone example are definitely different - they come from different warehouses, and so are physically different atoms. Similarly, Plates are physically different atoms. 

On a content side, if you are abstracting to the level of `Headphones` and not `Atoms making Headphones`, you're still getting the same parts, so each distribution would have the same contents.

> okay, so should we not be importing distribution ids or names then? How does one import a distribution without keeping record of the name and uuid?

I think we should be importing distribution ids and names. I was just pointing out that your personal node can just copy the distribution id and name and then change the contents, then advertise it has that distribution. This would be similar to counterfeit headphones.

Reply:
okay added a custom page for this.